### PR TITLE
Add initial canopyjs source

### PR DESCRIPTION
**Pull request by zapplebee**
The first of the APIs has been implemented in a minimal fashion

The notify method is the only exposed.

### CODE REVIEW

File: canopy/canopyjs/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      "es6": true,
+      "jest": true,
+      "browser": true
+    },
+    "rules": {
+      "import/no-default-export": 2,
+      "import/prefer-default-export": 0
+    },
+    "extends": ["plugin:@typescript-eslint/recommended", "airbnb", "plugin:prettier/recommended"]
+  }
```
Comment (dplumb94) [inline canopy/canopyjs/.eslintrc]: Add newline to end of file

File: canopy/canopyjs/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "printWidth": 80,
+    "tabWidth": 2,
+    "singleQuote": true
+  }
```
Comment (dplumb94) [inline canopy/canopyjs/.prettierrc]: Add newline to end of file

File: canopy/canopyjs/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      "eslint-plugin-import": "^2.18.2",
+      "eslint-plugin-jsx-a11y": "^6.2.3",
+      "eslint-plugin-prettier": "^3.1.1",
+      "eslint-plugin-react": "^7.16.0",
+      "jest": "^24.9.0",
+      "prettier": "^1.18.2",
+      "ts-jest": "^24.1.0",
+      "typescript": "^3.6.4"
+    }
+  }
```
Comment (dplumb94) [inline canopy/canopyjs/package.json]: Add newline to end of file

File: canopy/canopyjs/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        "esModuleInterop": true
+    },
+    "exclude": [
+        "node_modules",
+        "**/*.spec.ts"
+    ],
+    "include": [
+        "src/**/*"
+    ]
+}
```
Comment (dplumb94) [inline canopy/canopyjs/tsconfig.json]: Add newline to end of file

File: canopy/canopyjs/.eslintrc (hunk 1 above)
Comment (dplumb94) [inline canopy/canopyjs/.eslintrc]: It looks like some of the tabs are off in this file and the other JSON files

Comment (zapplebee) [reply canopy/canopyjs/.eslintrc]: Added the correct extensions to the format script in package.json

**Comment by zapplebee** [comment]
@dplumb94 I've still got changes requested from you, but those should be resolved. If you ok, I'll merge.

**Pull request merged**